onadata/apps/main/management/commands/update_enketo_urls.py

```diff
@@ -45,26 +48,35 @@ def handle(self, *args, **options):
 
         resultset = MetaData.objects.filter(
             Q(data_type='enketo_url') | Q(data_type='enketo_preview_url'))
+
         for meta_data in resultset:
-            username = meta_data.content_object.user.username
-            id_string = meta_data.content_object.id_string
+            username = None
+            id_string = None
+
+            if use_username:
+                username = meta_data.content_object.user.username
+                id_string = meta_data.content_object.id_string
+
             data_type = meta_data.data_type
             data_value = meta_data.data_value
             xform = meta_data.content_object
-            with open('/tmp/enketo_url', 'a') as f:
+            xform_pk = xform.pk
+            xform_uuid = xform.uuid if not use_username else None
```

Comment:
Not sure, If I understood this correctly. But I've changed this line to

```
xform_uuid = xform.uuid if not use_username and xform.uuid else None
```

Reply:
I'm suggesting to change to 

```python
xform_uuid = not use_username and xform.uuid
```

since this will evaluate to `False` when `useusername` is false, which is the correct behavior, we do not need to explicitly set `None` here

Reply:
The `xform_uuid` also acts as the container for the value (xform.uuid). We currently don't use this as a boolean.

Reply:
I understand, but this would not evaluate to a boolean if `not use_username` is true, it would be the intended value, no?

Reply:
My bad, yes this would be fine. Tested it out returns `False` when `not use_username` is False but that shouldn't be an issue as the functions will handle the value correctly either way.

I'll change this.

Reply:
Changed in the latest commits.